node/bft/src/sync/mod.rs

```diff
@@ -97,13 +103,36 @@ impl<N: Network> Sync<N> {
         self.sync_storage_with_ledger_at_bootup().await
     }
 
+    /// Performs one iteration of the block synchronization.
+    #[inline]
+    pub async fn try_block_sync(&self) {
+        // Prepare the block requests, if any.
+        // In the process, we update the state of `is_block_synced` for the sync module.
+        let (block_requests, sync_peers) = self.block_sync.prepare_block_requests();
+        trace!("Prepared {} block requests", block_requests.len());
+
+        // Process the block requests.
+        for requests in block_requests.chunks(DataBlocks::<N>::MAXIMUM_NUMBER_OF_BLOCKS as usize) {
+            if !self.block_sync.send_block_requests(&self.gateway, &sync_peers, requests).await {
+                // Stop if we fail to process a batch of requests.
+                break;
+            }
+
+            // Sleep to avoid triggering spam detection.
+            tokio::time::sleep(BLOCK_REQUEST_BATCH_DELAY).await;
+        }
+    }
+
     /// Starts the sync module.
+    ///
+    /// When this function returns sucessfully, the sync module will have spawned background tasks
+    /// that fetch blocks from other validators.
     pub async fn run(&self, sync_receiver: SyncReceiver<N>) -> Result<()> {
         info!("Starting the sync module...");
 
         // Start the block sync loop.
         let self_ = self.clone();
-        self.handles.lock().push(tokio::spawn(async move {
+        self.spawn(async move {
```

Comment:
I don't actually know the details of these handles/locks, but I was wondering why this change?

Reply:
`fn spawn(fut)` takes a future and calls `self.handles.lock().push(tokio::spawn(fut)))`.
The idea is that handles contains all background tasks started by the Sync module.

Strangely, this one line did not use the helper function. I assume it was just an oversight and changed it.